modules/core/src/pca.cpp

```diff
@@ -158,15 +158,14 @@ void PCA::write(FileStorage& fs ) const
     fs << "mean" << mean;
 }
 
-void PCA::read(const FileNode& fs)
+void PCA::read(const FileNode& fn)
 {
-    CV_Assert( !fs.empty() );
-    String name = (String)fs["name"];
-    CV_Assert( name == "PCA" );
+    CV_Assert( !fn.empty() );
+    CV_Assert( String(fn["name"]) == "PCA" );
```

Comment:
It is better to explicitly cast to _String_ to avoid ambiguity: `(String)fn["name"] == "PCA"`